### ISSUE TITLE
format command always fails when invoked directly from cli

### ISSUE THREAD

**Issue by GiuseppeIII** [biomejs/biome#5850]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.6.5

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         REVOKED
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0

For some additional context this is in wsl2
```

### What happened?

Running `pnpx @biomejs/biome@2.0.0-beta.2 format` gives

```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/node-semver-2.2.0/src/range.rs:982:18
Thread Name: biome::workspace_worker_13
Message: internal error: entered unreachable code: This should not have parsed: ~0.x.0

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/node-semver-2.2.0/src/range.rs:982:18
Thread Name: biome::workspace_worker_14
Message: internal error: entered unreachable code: This should not have parsed: ~0.x.0

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/node-semver-2.2.0/src/range.rs:982:18
Thread Name: biome::workspace_worker_9
Message: internal error: entered unreachable code: This should not have parsed: ~0.x.0

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/node-semver-2.2.0/src/range.rs:982:18
Thread Name: biome::workspace_worker_18
Message: internal error: entered unreachable code: This should not have parsed: ~0.x.0

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/node-semver-2.2.0/src/range.rs:982:18
Thread Name: main
Message: internal error: entered unreachable code: This should not have parsed: ~0.x.0
```

This occurs even in the simplest possible case against a single empty file
```
pnpx @biomejs/biome@2.0.0-beta.2 format test.js
```

### Expected result

To not error

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by nhedger** [comment]
Thanks for the report. Should be fixed by https://github.com/biomejs/biome/pull/5829

**Comment by nvsd** [comment]
Is there a way to use this today before an official release?

**Comment by ematipico** [comment]
> Is there a way to use this today before an official release?

https://github.com/biomejs/biome/blob/main/RELEASES.md
